wire-runtime/src/main/java/com/squareup/wire/WireAdapter.java

```diff
@@ -347,6 +334,27 @@ public String toString(E value) {
     }
   };
 
+  private static final Map<WireType, WireAdapter<?>> TYPE_TO_ADAPTER;
+  static {
+    Map<WireType, WireAdapter<?>> map = new LinkedHashMap<WireType, WireAdapter<?>>();
+    map.put(WireType.BOOL, WireAdapter.BOOL);
```

Comment:
tempting follow up: API on WireType to return a TypeAdapter if it is a scalar. Trick is what it does otherwise.